tests/foreman/cli/test_leapp_client.py

```diff
@@ -0,0 +1,344 @@
+    for rh_repo_key in RHEL_REPOS.keys():
+        release_version = RHEL_REPOS[rh_repo_key]['releasever']
+        if release_version in str(source) or release_version in target:
+            # prod = RHEL_REPOS[rh_repo_key]['product'] #rh_repo_key.split('_')[0]
+            prod = 'rhel' if 'rhel7' in rh_repo_key else rh_repo_key.split('_')[0]
+            if module_stash[synced_repos].get(rh_repo_key, None):
+                logger.info("Repo %s already synced, not syncing it", rh_repo_key)
+            else:
+                logger.info('Enabling %s repository in product %s', rh_repo_key, prod)
+                module_stash[synced_repos][rh_repo_key] = True
+                repo_id = module_target_sat.api_factory.enable_rhrepo_and_fetchid(
+                    basearch=custom_leapp_host.arch,
+                    org_id=module_sca_manifest_org.id,
+                    product=PRDS[prod],
+                    repo=RHEL_REPOS[rh_repo_key]['name'],
+                    reposet=RHEL_REPOS[rh_repo_key]['reposet'],
+                    releasever=release_version,
+                )
+                rh_repo = module_target_sat.api.Repository(id=repo_id).read()
+                all_repos.append(rh_repo)
+                logger.info('Syncing %s repository', rh_repo_key)
+                # sync repo
+                rh_repo.sync(timeout=1800)
+    function_leapp_cv.repository = all_repos
+    function_leapp_cv = function_leapp_cv.update(['repository'])
+    logger.info('Repos to be added to the AK: %s', all_repos)
+    # Publish, promote content view to lce
+    logger.info(f'Publish, promote cv - {function_leapp_cv.name} to lce - {module_leapp_lce.name}')
+    function_leapp_cv.publish()
+    cvv = function_leapp_cv.read().version[0]
+    cvv.promote(data={'environment_ids': module_leapp_lce.id, 'force': True})
+    function_leapp_cv = function_leapp_cv.read()
+
+
+@pytest.fixture
+def custom_leapp_host(upgrade_path):
+    deploy_args = {}
+    deploy_args['deploy_rhel_version'] = upgrade_path['source_version']
+    logger.info('Creating Leapp Host - RHEL %s', deploy_args)
+    with Broker(
+        workflow='deploy-rhel',
+        host_class=ContentHost,
+        deploy_rhel_version=upgrade_path['source_version'],
+        deploy_flavor=settings.flavors.default,
+    ) as chost:
+        yield chost
+
+
+@pytest.mark.parametrize(
+    'upgrade_path',
+    [
+        # {'source_version': RHEL7_VER, 'target_version': RHEL8_VER},
+        {'source_version': RHEL8_VER, 'target_version': RHEL9_VER},
+    ],
+    ids=lambda upgrade_path: f'{upgrade_path["source_version"]}'
+    f'_to_{upgrade_path["target_version"]}',
+)
+@pytest.mark.usefixtures(
+    'leapp_sat_content',
+    'register_host_with_satellite',
+    'verify_target_repo_on_satellite',
+    'precondition_check_upgrade_and_install_leapp_tool',
+    'fix_inhibitors',
+)
+def test_leapp_upgrade_rhel(
+    module_target_sat,
+    custom_leapp_host,
+    upgrade_path,
```

Comment:
- Initially `@pytest.mark.usefixtures(...)` included in test module to execute fixtures in expected order. 
- We want some fixtures (pre-requisites) to be executed prior to run test case
- I will work on suggested changes, if suggested changes work as expected then I will include in upcoming commit otherwise will keep as it is.

Reply:
I, personally see no value and reason in putting fixtures whose return value is not used in the function body to the function arguments. IMO using the `usefixtures` marker makes the code more concise and readable.

There are a few usages of this marker in robottelo and I personally believe that it is a better practice to use the `usefixtures` marker instead of putting the fixture to function arguments if its return value is not used in the test.